test/unittest/blas1/blas1_sdsdot_test.cpp

```diff
@@ -0,0 +1,131 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Dotright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a dot of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a dot of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_sdsdot_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+#include <iostream>
+#include <type_traits>
+
+enum class api_type : int { event = 0, result = 1 };
+
+/**
+ * Operator overload to print the test name properly
+ */
+std::ostream& operator<<(std::ostream& os, const api_type& type) {
+  if (type == api_type::event) {
+    return os << "event";
+  } else {
+    return os << "result";
+  }
+}
+
+template <typename scalar_t>
+using combination_t = std::tuple<api_type, int, float, int, int>;
```

Comment:
```
template <typename scalar_t>
using combination_t = std::tuple<api_type, int, scalar_t, int, int>;
```
?

Reply:
Fair enough. Will update it

Reply:
Done

Reply:
Does the test now compile / pass with DPC++? It is currently ComputeCpp only in the unittest cmake.

Reply:
yes, updated it again to enable the test on dpcpp